docker/base/elixir-cuda.dockerfile

```diff
@@ -1,11 +1,12 @@
 ARG ELIXIR_VERSION
 ARG ERLANG_VERSION
-
 ARG CUDA_VERSION
+ARG ARCH
+ARG BASE_IMAGE=hexpm/elixir:${ELIXIR_VERSION}-erlang-${ERLANG_VERSION}-ubuntu-focal-20211006
 
-FROM hexpm/elixir:${ELIXIR_VERSION}-erlang-${ERLANG_VERSION}-ubuntu-focal-20211006 AS elixir
+FROM --platform=linux/${ARCH} ${BASE_IMAGE} AS elixir
```

Comment:
It is funny you brought up `docker buildx` (in the linked article) I saw it while building out this PR and wondered if it would be useful here. But I wanted to keep up with the convention. Though I didn't fully understand the whole build pipeline.